### ISSUE TITLE
`ServiceList` Does Not Preserve Keys

### ISSUE THREAD

**Issue by XedinUnknown** [Dhii/services#21]
The `ServiceList` class is very useful for combining several services into a list. One of its useful abilities is to map them to keys that have meaning, rather than just the numeric index in the service definition list.

Unfortunatley, the mapping doesn't seem to work: it [uses](https://github.com/Dhii/services/blob/0.1.x/src/Factories/ServiceList.php#L60) the olde `resolveDeps()`, despite `use`ing the `ResolveKeysCapableTrait`.

Moreover, `ResolveKeysCapableTrait#resolveKeys()`, which should presulably be used to preserve the keys, uses `resolveDeps()` directly, without changing anything. And that, as we know, does not preserve keys.

## Suggested Solution
It seems that the `ResolveKeysCapableTrait` was added in order to distinguish between key-knowing and key-agnostic resolution. Probably, just forgot to change `ResolveKeysCapableTrait::resolveDeps()` to use the key rather than `[]` (push).

**Comment by mecha** [comment]
I don't think we forgot to update `resolveDeps()`. Rather, I think we forgot to update `ServiceList`.

See [here](https://github.com/Dhii/services/pull/7#issuecomment-867437677).
